tests/pytest.org.md

```diff
@@ -0,0 +1,122 @@
+# Pytest organization proposal
+
+This proposal intends to achieve the following
+  - Have a standard way of categorizing tests
+  - Have some guidelines around test file organization
+  - Have a master wrapper for test execution
+  - Follow common documentation style
+  - Test result collection
+
+## Test categorization
+Leverage pytest custom markers to group tests based on topology, platform, features and connection type.
+
+```
+pytest.ini
+[pytest]
+markers:
+    topology(topo_name): The topologies this particular testcase can run against. topo_name can be individual topology names like 't0', 't1', 'ptf', 'any' or a comma separated like ('t0', 't1') if supported on multiple topologies
+    platform(platform_name): used for platform specific test(broadcom, mellanox, vs etc)
+    feature(feature_name): feature this test is written for. eg. acl, nat
+    connection_type(name): names can be 'fabric' (which indicates the presence of a fanout switch) or 'direct' if a testcase uses directly connected links
+
+```
+conftest.py
+
+```
+def pytest_addoption(parser):
+        parser.addoption("--topology", action="store", metavar="NAME",
+                        help="only run tests matching the topology NAME")
+
+def pytest_runtest_setup(item):
+    toponames = [mark.args for mark in item.iter_markers(name="topology")]
+    if toponames:
+        cfg_topos = item.config.getoption("--topology").split(',')
+        if all(topo not in toponames[0] for topo in cfg_topos):
+            pytest.skip("test requires topology in {!r}".format(toponames))
+    else:
+        if item.config.getoption("--topology"):
+            pytest.skip("test does not match topology")
+
```

Comment:
Did we end up deciding that un-marked tests should not run if a topology is specified? Could we describe the expected behavior/rationale somewhere in the docs so we have it documented?

Reply:
My thought is there wouldn't be any tests eventually in the repo that aren't marked with the topology tag. If there are any, it would be because the author missed marking it. I think in such cases, it would be safer to leave them out rather than attempting to run them on some topology they aren't expected to run and pollute our results. 

I can add a comment in there to make it more explicit